lib/discourse_plugin_registry.rb

```diff
@@ -15,7 +17,7 @@ class DiscoursePluginRegistry
   #   - Defines instance method as a shortcut to the singleton method
   #   - Automatically deletes the register on registry.reset!
   def self.define_register(register_name, type)
-    @@register_names ||= Set.new
+    return if respond_to?(register_name)
```

Comment:
Small optimization, if those methods are called more than once for an existing register, they won’t recreate everything eah time.

Reply:
Alternatively you could `@@register_names.add?(register_name)` and call the `define_*_method` if a value is returned (meaning it has been added to the set)

Reply:
My first implementation was to check if the value was already in the set, but this was breaking the specs because at some point the class variable is kept while the methods were removed. I experimented a bit with an instance class variable, but there were other problems. As this is a relatively important piece of code and I didn’t want to spend too much time on it, I found that solution :grin: